pkg/volume/configmap/configmap.go

```diff
@@ -213,6 +207,13 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		}
 	}
 
+	if err := wrapped.SetUpAt(dir, fsGroup); err != nil {
+		return err
+	}
+	if err := volumeutil.MakeNestedMountpoints(b.volName, dir, b.pod); err != nil {
```

Comment:
My concern is that any of the steps after this can fail and we won't clean up the result of this method. Maybe we can add a clean up routine to execute if any of the subsequent steps fail?

Reply:
Ditto for other plugins.

Reply:
Yeah - so as I commented above, I omitted that case on purpose and there were 2 reasons of that:

1. It is much harder to test other failures and hence much smaller chance of them happening in real world. 
2. cleaning up after both wrapped emptydir and nested mount points has been created is  messier and since configmap/secrets are such core k8s constructs, I want to approach that cautiously. Problem is - first we have to remove the mount points we created and then unmount emptydir and I am worried we do not have enough test coverage around that area. :(

Reply:
Ok, I guess this will take care of the worst cases. So it is better then nothing.